### PR TITLE
Promisified the `request` method

### PR DESCRIPTION
**Pull request by avrtau**
FEATURE: request method now returns a Promise if no callback is specified.

BREAKING CHANGE: Callback for makeRequest parameter change from (error, result, headers) to (error, {result, headers}).

Documentation is updated

### CODE REVIEW

**Comment by ericblade** [comment]
Hey @avrtau , I updated the tests to match the new API, added a test for Promise, bumped version, and put you into the Contributors list.  I'd appreciate it if you'd cherry-pick this change, or write something similar yourself, and attach it to this pull request.  thanks!

https://github.com/ericblade/mws-simple/commit/3e6163804fcd55bdbb80cd74299314cf1e901152

**Comment by ericblade** [comment]
oh, also, the reason why coverage is complaining is because travis-ci for some reason does not provide API credentials unless I do the pull request.  So don't worry about that, I always manually run tests before i accept anything :)

**Comment by avrtau** [comment]
> Hey @avrtau , I updated the tests to match the new API, added a test for Promise, bumped version, and put you into the Contributors list. I'd appreciate it if you'd cherry-pick this change, or write something similar yourself, and attach it to this pull request. thanks!
> 
> [3e61638](https://github.com/ericblade/mws-simple/commit/3e6163804fcd55bdbb80cd74299314cf1e901152)

I am not 100% sure what you are asking me to do - not very familiar with github yet. Should I fork the newest version and add a pull request there?

**Comment by ericblade** [comment]
@avrtau probably quickest to do would be 

````
git remote add ericblade git@github.com:ericblade/mws-simple.git
git fetch ericblade
git cherry-pick 3e61638
git push origin master
````

that should take my commit 3e61638 and put it into your pull req.  That way the fixes to the test go into the pull request together with the changes, instead of having them separate.  :)

**Comment by avrtau** [comment]
> @avrtau probably quickest to do would be
> 
> ```
> git remote add ericblade git@github.com:ericblade/mws-simple.git
> git fetch ericblade
> git cherry-pick 3e61638
> git push origin master
> ```
> that should take my commit [3e61638](https://github.com/ericblade/mws-simple/commit/3e6163804fcd55bdbb80cd74299314cf1e901152) and put it into your pull req. That way the fixes to the test go into the pull request together with the changes, instead of having them separate. :)

when I fetch:
`git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.`

**Comment by avrtau** [comment]
codebeat is relentless

**Comment by ericblade** [comment]
Don't worry about codebeat :-) or the coverage warnings.  Codebeat isn't very configurable, and most of what it warns about aren't really worth worrying about (very low on the error threshold)

Thanks for updating the pull!

File: mws-simple.js [hunk 1]
```diff
@@ -6,25 +6,26 @@ const getContentType = require('./lib/getContentType');
...
+    });
+}
+
 class MWSSimple {
     constructor({ appId = pkgAppId, appVersionId = pkgAppVersionId, host = 'mws.amazonservices.com', port = 443, accessKeyId, secretAccessKey, merchantId, authToken } = {}) {
         Object.assign(this, { appId, appVersionId, host, port, accessKeyId, secretAccessKey, merchantId, authToken, ServerError });
         
         // allows to use this inside the request method
-        this.request = this.request.bind(this);
+        promisify = promisify.bind(this);
```
Comment (ericblade) [inline mws-simple.js]: So, this fails pretty hard now, after this change.  (npm test fails the Promise check, as the server receives the wrong credentials) 

I think you could use node 8's 'utils' or 'util' lib that has a built-in promisify, and just make sure to do the this.request.bind(this) in constructor.  As is, putting promisify outside the class, doesn't work, as every time you construct a new class instance, it's changing the definition of promisify inside the module, to be bound to a different instance.

Comment (ericblade) [reply mws-simple.js]: If you have a MWS login, please put your keys in test/keys.json, and run npm test . make sure you don't commit that file, though. :-)  The test for Promise usage fails, as it checks the server return value, expecting "Not Found" but instead receives "Not Authorized", as 'this' is still pointing at the fake credentials made in the mws-simple defaults test

**Pull request merged**